app/views/root/_gem_base.html.erb

```diff
@@ -41,7 +28,7 @@
   account_nav_location: account_nav_location,
   **(defined?(blue_bar) ? {blue_bar: blue_bar,} : {}),
   draft_watermark: draft_environment,
-  emergency_banner: emergency_banner.presence,
+  emergency_banner: render("govuk_web_banners/emergency_banner", homepage:),
```

Comment:
Just curious... how does the text for the banner get passed to it?

Reply:
The emergency banner work by a shared Redis cluster. There's a UI in Whitehall that writes a record to the cluster (including the text), and when you call this partial, the partial makes a call to Redis to check if the record has any values. If yes, the partial renders the component. If not, the partial renders empty. . (call is cached for 1 minute so we're not constantly hammering the redis instance).